Formula/libngspice.rb

```diff
@@ -0,0 +1,79 @@
+class Libngspice < Formula
+  desc "Spice circuit simulator as shared library"
+  homepage "http://ngspice.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/ngspice/ng-spice-rework/26/ngspice-26.tar.gz"
+  sha256 "51e230c8b720802d93747bc580c0a29d1fb530f3dd06f213b6a700ca9a4d0108"
+
+  head do
+    url "git://git.code.sf.net/p/ngspice/ngspice"
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "bison" => :build
+    depends_on "flex" => :build
+    depends_on "libtool" => :build
+
+    # Currently, headers don't get installed to include/*.
+    # There is a patch upstream that addresses this for HEAD.
+    # Upstream ticket: https://sourceforge.net/p/ngspice/bugs/327
+    patch :DATA
+  end
+
+  option "without-cider", "Build without CIDER numerical device simulator"
+  option "without-xspice", "Build without XSPICE extensions"
```

Comment:
I'm sorry for force-pushing here. Next time, I will just add another commit. I am still aware of the unresolved issue that was brought up with the two options.

The last comment on this by @bfontaine was:

> Why would someone install `libngspice` without them? (i.e. are these options really needed?)

and I answered:

> I think this depends on the specific use case that people have in mind with using the library. I have no problem with hard-coding them in, but I thought that offering them as options would be most convenient, i.e. if people want to disable them they can do so rather than being forced to a specific set of options.